HCollections.Test/TestHListT.fs

```diff
@@ -35,3 +34,42 @@ module TestHListT =
             |> HListT.cons 4.5 10
 
         Assert.Equal (2, HListT.length testHlist)
+
+    [<Fact>]
+    let ``HlistT.tolist returns the correct list`` () =
+        let testHlist =
+            HListT.empty<int>
+            |> HListT.cons "hi" 4
+            |> HListT.cons 4.5 10
+            |> HListT.toList
+
+        Assert.Equal<int list> ([10; 4], testHlist)
+
+    [<Fact>]
+    let ``HListT.toList on an empty HListT returns an empty list`` () =
+        let testList = HListT.toList HListT.empty<int>
+
+        Assert.Equal<int list> ([], testList)
+
+    [<Fact>]
+    let ``HListT.toHList returns the correct HList type`` () =
```

Comment:
Not a huge fan of how I'm testing the HList component. However, in order to do it properly I'd need to add an equality function to the library (or maybe the DU internal rather than private and then write something in the tests). Happy to do that if you agree.

Reply:
I don't really want an equality function in the library, because then the type constraints would become nightmarish (what if you want an HList whose types are constrained only to be equatable? we should only do that after some thought). `toTypes` is absolutely fine.

Reply:
Yeah, my personal preference would have been to create a custom equality checker in the test file. At the moment the test would pass if I just supplied Unchecked.defaultof but making massive changes isn't a price I wanted to pay unless we were in agreement.

Reply:
Yes, I agree that trying to define equality would be a bit of a nightmare - these collections have `[<NoComparison>]` and `[<NoEquality>]` attributes for a reason! 😄